allensdk/brain_observatory/filtered_running_speed/__main__.py

```diff
@@ -0,0 +1,954 @@
+            np.logical_and(vsig < min_threshold, shifted_vsig > max_threshold)
+            ).nonzero()[0]
+        neg_wraps = np.asarray(
+            np.logical_and(vsig > max_threshold, shifted_vsig < min_threshold)
+            ).nonzero()[0]
+
+    return pos_wraps, neg_wraps
+
+
+def extract_running_speeds(
+    dx_raw,
+    frame_times,
+    vsig,
+    vin,
+    start_index,
+    end_index,
+    lowpass: bool = True,
+    zscore_threshold=DEFAULT_ZSCORE_THRESHOLD,
+):
+    """
+    Given the dx_deg from the 'pkl' file object and a 1d
+    array of timestamps, compute the running speed. Returns a
+    dataframe with the raw voltage dx_deg as well as the computed speed
+    at each timestamp. By default, the running speed is filtered with
+    a 10 Hz Butterworth lowpass filter to remove artifacts caused by
+    the rotary encoder.
+
+    Parameters
+    ----------
+    dx_raw
+        Deserialized 'pkl' file dx_deg
+    frame_times: np.ndarray (1d)
+        Timestamps for running data measurements
+    vsig: np.ndarray (1d),
+        raw analog data for encoder
+    vin: np.ndarray (1d),
+        input voltage to the encoder
+    start_index: int,
+        the start index of the stimulus in reference to the full vsync times
+    end_index: int,
+        the end index of the stimulus in reference to the full vsync times
+    lowpass: bool (default=True)
+        Whether to apply a 10Hz low-pass filter to the running speed
+        data.
+    zscore_threshold: float
+        The threshold to use for removing outlier running speeds which might
+        be noise and not true signal.
+
+    Returns
+    -------
+    pd.DataFrame
+        Dataframe with an index of timestamps and the following
+        columns:
+            "velocity": computed running speed
+            "net_rotation": dx in radians
+            "frame_indexes": frame indexes into the raw frame time,
+            "frame_times": the frame times for the timestamp data
+
+    Notes
+    -----
+    Though the angular change is available in the raw data
+    (key="dx"), this method recomputes the angular change from the
+    voltage signal (key="vsig") due to very specific, low-level
+    artifacts in the data caused by the encoder. See method
+    docstrings for more detailed information. The raw data is
+    included in the final output in case the end user wants to apply
+    their own corrections and compute running speed from the raw
+    source.
+    """
+
+    # Identify "wraps" in the voltage signal that need to be unwrapped
+    # This is where the encoder switches from 0V to 5V or vice versa
+    pos_wraps, neg_wraps = _identify_wraps(
+        vsig, min_threshold=1.5, max_threshold=3.5)
+
+    # Unwrap the voltage signal and apply correction for transient spikes
+    unwrapped_vsig = _unwrap_voltage_signal(
+        vsig, pos_wraps, neg_wraps, max_threshold=5.1, max_diff=1.0)
+
+    angular_change_point = _angular_change(unwrapped_vsig, vin)
```

Comment:
The docstring of `_angular_change` refers to the second argument as the "maximum voltage allowed on the encoder". Here you are passing the "input voltage on the encoder." Are these the same thing? Can you add a comment explaining why those two things are equivalent?

Reply:
This has been removed from my code as I am now calling a library that contains this.